sources/packages/web/src/components/common/IdleTimeChecker.vue

```diff
@@ -57,44 +84,114 @@ export default {
     const resetIdleCheckerTimer = () => {
       clearInterval(interval.value);
       if (isAuthenticated.value) {
-        /* eslint-disable */
-        interval.value = setInterval(checkIdle, 30000);
-        /* eslint-enable */
+        interval.value = setInterval(checkIdle, 1000);
       }
     };
 
+    watch(
+      () => modalOpen.value,
+      (newValue: boolean) => {
+        if (newValue) {
+          countdownInterval.value = setInterval(() => {
+            if (countdown.value > 0) {
+              countdown.value = maximumIdleTime.value - idleTimeInSeconds.value;
+            }
+          }, 1000);
+        } else {
+          clearInterval(countdownInterval.value);
+          countdown.value = COUNT_DOWN_TIMER_FOR_LOGOUT;
+        }
+      },
+    );
+
+    const extendUserSessionTime = () => {
+      setLastActivityTime();
+      resetIdleCheckerTimer();
+      modalOpen.value = false;
+      clearInterval(countdownInterval.value);
+      countdown.value = COUNT_DOWN_TIMER_FOR_LOGOUT;
+    };
+
     const confirmExtendTimeModal = async () => {
+      modalOpen.value = true;
       if (await extendTimeModal.value.showModal()) {
-        lastActivityLogin.value = new Date();
-        resetIdleCheckerTimer();
+        extendUserSessionTime();
+      } else {
```

Comment:
Can we please remove the else here?

Reply:
No, we can't. If the condition returns false, it means that the user clicked on "No" to extend their user session time. In other words, the user wants to be logged out.